opennms-services/src/main/java/org/opennms/netmgt/poller/pollables/PollContext.java

```diff
@@ -78,10 +79,10 @@
      * @param address a {@link java.net.InetAddress} object.
      * @param svcName a {@link java.lang.String} object.
      * @param date a {@link java.util.Date} object.
-     * @param reason a {@link java.lang.String} object.
+     * @param params a {@link java.util.Map} object containing {@link java.ling.String} key to {@link java.ling.String} value mappings.
      * @return a {@link org.opennms.netmgt.xml.event.Event} object.
      */
-    public Event createEvent(String uei, int nodeId, InetAddress address, String svcName, Date date, String reason);
+    public Event createEvent(String uei, int nodeId, InetAddress address, String svcName, Date date, Map<String,String> params);
```

Comment:
Can we just add a new method that takes a Map here instead? That way the existing calls can stay untouched.